rpc-broker/src/rpc-broker.h

```diff
@@ -0,0 +1,368 @@
+    struct sockaddr_in peer;
+};
+
+#define WS_LOOP_TIMEOUT             100  // length of time each service of the websocket
+                                         // event-loop (millisecs)
+#define DBUS_BROKER_CLIENT_TIMEOUT  100
+#define DBUS_BROKER_MSG_TIMEOUT     100
+
+// DBus-Broker messages
+#define DBUS_REQ_TIMEOUT   10000
+#define DBUS_MSG_LEN        8192
+#define DBUS_ARG_LEN        1024
+
+#define DBUS_INTROSPECT_MAX 0xFFFF
+
+// The actual length is 32, but sockaddr_un.sun_path is 108 chars maximum
+#define DBUS_BUS_ADDR_LEN   108
+
+#define DBUS_MAX_ARG_LEN    16
+
+struct dbus_message {
+    const char *destination;
+    const char *interface;
+    const char *path;
+    const char *member;
+    const char *type;
+    // struct argument {
+    //   void *data;
+    //   char dbus_type;
+    //   char actual_type;
+    // };
+    uint8_t arg_number;
+    char arg_sig[DBUS_MAX_ARG_LEN];
+    char json_sig[DBUS_MAX_ARG_LEN];
+    void *args[DBUS_MAX_ARG_LEN];
+};
+
+#define DBUS_READ "read"
+#define DBUS_LIST "list"
+
+#define CLIENT_REQ_LEN       16
+#define CLIENT_DBUS_REQ_LEN 256
+
+#define VM_UUID_LEN   33
+#define DOMID_UUID_LEN 8
+#define DOMID_SECTION 24
+
+#define XENSTORE_TARGET_LEN 8
+#define XENSTORE_TARGET "/target"
+
+#define WS_RING_BUFFER_MEMBER_SIZE 8192
+#define WS_RING_BUFFER_MEMBER_NUM    64
+
+#define WS_USER_MEM_SIZE 8192  // the amount memory that is allocated for user
+                               // for each ws-callback
+
+struct json_request {
+    uint32_t id;
+    DBusConnection *conn;
+    struct lws *wsi;
+    struct dbus_message dmsg;
+};
+
+#define JSON_REQ_ID_MAX 16
+#define JSON_TYPE_MAX   16
+#define JSON_REQ_MAX   256
+
+struct json_response {
+    uint32_t id;
+    const char *path;
+    const char *interface;
+    const char *member;
+    char response_to[JSON_REQ_ID_MAX];
+    char type[JSON_TYPE_MAX];
+    char arg_sig[DBUS_MAX_ARG_LEN];
+    struct json_object *args;
+};
+
+static const char json_dbus_types[] = {
+    [json_type_boolean] = 'b',
+    [json_type_double]  = 'd',
+    [json_type_int]     = 'i',
+    [json_type_string]  = 's'
+};
+
+#define XML_SIGNATURE_MAX 16
+
+static const xmlChar XML_NAME_PROPERTY[]      = "name";
+static const xmlChar XML_DIRECTION_PROPERTY[] = "direction";
+
+static const char XML_IN_FIELD[]           = "in";
+static const char XML_TYPE_FIELD[]         = "type";
+
+struct lws_ring *ring;
+bool verbose_logging;
+int dbus_broker_running;
+struct policy *dbus_broker_policy;
+bool reload_policy;
+bool dom0;
+
+
+struct dbus_broker_args {
+    bool logging;
+    bool verbose;
+    int port;
+    const char *bus_name;
+    const char *logging_file;
+    const char *rule_file;
+};
+
+struct raw_dbus_conn {
+    int server;
+    int client;
+    uint32_t client_domain;
+    uv_poll_t handle;
+};
+
+// rpc-broker.c
+void broker_message(struct raw_dbus_conn *rdconn);
+
+signed int is_stubdom(uint16_t domid);
+
+void print_usage(void);
+
+void service_rdconns(void);
```

Comment:
Function does not exist, is this supposed to be the declaration for service_rdconn_cb?